src/test/ui/issues/issue-73427.stderr

```diff
@@ -4,39 +4,65 @@ error[E0423]: expected value, found enum `A`
 LL |     A.foo();
    |     ^
    |
-   = help: you might have meant to use one of the enum's other variants that have fields
-help: try using one of the enum's variants
+help: try using one of the enum's variants without fields
    |
 LL |     (A::Struct {}).foo();
    |     ^^^^^^^^^^^^^^
 LL |     (A::Tuple()).foo();
    |     ^^^^^^^^^^^^
 LL |     A::Unit.foo();
    |     ^^^^^^^
+help: try using one of the enum's variants with fields
```

Comment:
It might just be that I'm not a native English speaker, but the with/without fields wording feels weird to me. Also, I think I would personally prefer a single list of all variants instead of two here.

Reply:
I've improved the grammar in the message. I can't do a single list without having the same applicability - and in this case, both lists should have different applicabilities.

Reply:
I am ok with keeping two different lists, but would like to change the wording somewhat. I'll propose a wording change in a separate comment.